File: src/main/java/com/gangofconnectfour/powerfourservice/repository/impl/UserDetailsServiceImpl.java
package com.gangofconnectfour.powerfourservice.repository.impl;

import com.gangofconnectfour.powerfourservice.model.User;
import com.gangofconnectfour.powerfourservice.repository.UserRepository;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import java.util.Collections;

@Service
public class UserDetailsServiceImpl implements UserDetailsService {

    public UserService userService;

    public UserDetailsServiceImpl(UserService userService) {
        this.userService = userService;
    }

    @Override
    public UserDetails loadUserByUsername(String s) throws UsernameNotFoundException {
        User userFounded = this.userService.getUserByMail(s);
        if (userFounded == null)
            throw new UsernameNotFoundException(s);
        return new org.springframework.security.core.userdetails.User(userFounded.getEmail(), userFounded.getPassword(), Collections.emptyList());
    }
}

File: src/main/kotlin/com/gangofconnectfour/powerfourservice/repository/impl/UserDetailsServiceImpl.kt
package com.gangofconnectfour.powerfourservice.repository.impl

import com.gangofconnectfour.powerfourservice.repository.UserRepository
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class UserDetailsServiceImpl(applicationUserRepository : UserRepository) : UserDetailsService {

    var applicationUserRepository = applicationUserRepository

    override fun loadUserByUsername(username: String?): UserDetails {
        var applicationUser = applicationUserRepository.getUserByMail(username!!);
        if (applicationUser == null) {
            throw UsernameNotFoundException(username)
        }
        return org.springframework.security.core.userdetails.User(applicationUser.email, applicationUser.password, emptyList<GrantedAuthority>())
    }
}